ext/zip/php_zip.c

```diff
@@ -1795,16 +1811,7 @@ static ZIPARCHIVE_METHOD(addFromString)
 		RETURN_FALSE;
 	}
 
-	cur_idx = zip_name_locate(intern, (const char *)name, 0);
-	/* TODO: fix  _zip_replace */
-	if (cur_idx >= 0) {
-		if (zip_delete(intern, cur_idx) == -1) {
-			zip_source_free(zs);
-			RETURN_FALSE;
-		}
-	}
```

Comment:
Is removing this code intentional?

Reply:
Yes, as we now use `zip_file_add` (instead of deprecated `zip_add_file`), we can use the ZIP_FL_OVERWRITE option which does the same (previous behavior is preserve, but use can choose to set `flags=0`, and thus use a more strict behavior, see `tests/oo_add_from_string.phpt `)

Reply:
Ah, thanks for the explanation!